components/support/utils/src/main/java/mozilla/components/support/utils/DomainMatcher.kt

```diff
@@ -31,7 +31,7 @@ private fun basicMatch(query: String, urls: Sequence<String>): String? {
         }
 
         val url = try {
-            URL(rawUrl)
+            Uri.parse(rawUrl)
         } catch (e: MalformedURLException) {
             null
         }
```

Comment:
nit: It looks like this could bail out early and we do not need to continue with a null `Uri`.

Reply:
Actually it looks like `Uri` will never throw `MalformedURLException`. You just won't get values for certain getters.